src/Features/CSharp/Portable/InitializeParameter/CSharpAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -92,5 +92,22 @@ protected override StatementSyntax CreateParameterCheckIfStatement(DocumentOptio
                 statement: ifTrueStatement,
                 @else: null);
         }
+
+        protected override async Task<Document?> TryAddNullCheckToParameterDeclarationAsync(Document document, IParameterSymbol parameter, CancellationToken cancellationToken)
+        {
+            var tree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var options = (CSharpParseOptions)tree.Options;
+            if (options.LanguageVersion < LanguageVersion.Preview)
+            {
+                return null;
+            }
+
+            var syntaxRoot = await tree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+            var syntax = (ParameterSyntax)await parameter.DeclaringSyntaxReferences[0].GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
The node was already calculated in:

https://github.com/dotnet/roslyn/blob/d959aaf644ccc7f23f011cc4e8cadcf5b50133c9/src/Features/Core/Portable/InitializeParameter/AbstractInitializeParameterCodeRefactoringProvider.cs#L72

Would it be easy to pass it around instead of re-calculating?

Reply:
yes.  i'd prefer this as well, so all features have a consistent view of what syntax they're operating on (may be relevant in error cases).

Reply:
Done.